src/public/modules/forms/base/directives/field.client.directive.js

```diff
@@ -23,6 +28,27 @@ function fieldDirective(FormFields) {
       isValidateDate: '<',
     },
     link: function (scope) {
+      // Stealth prefill feature
+      // If a query parameter is provided to a form URL in the form ?<fieldId1>=<value1>&<fieldId2>=<value2>...
+      // And if the fieldIds are valid mongoose object IDs and refer to a short text field,
+      // Then prefill and disable editing the corresponding form field on the frontend
+
+      const queryParams = $location.search()
+      console.log(queryParams)
+
+      if (
+        scope.field._id in queryParams &&
+        scope.field.fieldType === 'textfield'
+      ) {
+        let prefillValue = queryParams[scope.field._id]
+        prefillValue = Array.isArray(prefillValue)
```

Comment:
it's more defensive if we test whether the prefill value is a string, since that is the only use case we would like to support